src/schemas/Helpdesk/jiraRequests.ts

```diff
@@ -0,0 +1,208 @@
+import { JIRA_ORGANIZATION_ID, JIRA_ADMIN_EMAIL, JIRA_ADMIN_PASS, JIRA_REST_URI, JIRA_SERVICEDESK_ID } from 'config';
+import { restErrorResponseHandler } from 'utils/restUtils';
+import fetch from 'node-fetch';
+import { UserInputError } from 'apollo-server-express';
+import { loadVaultSecret } from 'services/vault';
+import {
+  USE_VAULT,
+  JIRA_ADMIN_VAULT_CREDENTIALS_PATH,
+} from 'config';
+import logger from 'utils/logger';
+
+type JiraCredentials = {
+  email: string;
+  password: string;
+};
+
+const isCredentials = (data: { [k: string]: any }): data is JiraCredentials => {
+  return typeof data['email'] === 'string' && typeof data['password'] === 'string';
+};
+
+const getJiraBasicAuth = async (): Promise<string> => {
+  let basicAuthString: string;
+  if (USE_VAULT) {
+    const secretData = await loadVaultSecret()(JIRA_ADMIN_VAULT_CREDENTIALS_PATH).catch(err => {
+      logger.error(
+        `could not read Jira Credentials secret at path ${JIRA_ADMIN_VAULT_CREDENTIALS_PATH}`,
+      );
+      throw err;
+    });
+    if (isCredentials(secretData)) {
+      basicAuthString = `${secretData.email}:${secretData.password}`
+    } else {
+      throw new Error(`vault secret at ${JIRA_ADMIN_VAULT_CREDENTIALS_PATH} could not be read`);
+    }
+  } else {
+    basicAuthString = `${JIRA_ADMIN_EMAIL}:${JIRA_ADMIN_PASS}`
+  }
+
+  return `Basic ${Buffer.from(basicAuthString).toString('base64')}`
+};
+
+
+const getJiraRequestHeader = async () => {
+
+  const auth = await getJiraBasicAuth();
+  return {
+    Authorization: auth,
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  };
+}
+
+
+export const createServiceRequest = async (customerIdentifier: string, requestTypeId: string, requestText:string, summaryPrependText: string) => {
+
+    const url = `${JIRA_REST_URI}/request`;
+
+    const characterLimit = 60;
+    const summaryText = requestText.length > characterLimit ? `${requestText.substring(0,characterLimit)}...` : requestText;
+
+    // summary is a required field, that cannot accept newlines
+    const request = {
+      raiseOnBehalfOf: customerIdentifier,
+      serviceDeskId: JIRA_SERVICEDESK_ID,
+      requestTypeId: requestTypeId,
+      requestFieldValues: {
+        summary: `${summaryPrependText}: ${summaryText.replace(/(\r\n|\n|\r)/gm, "")}`,
+        description: requestText
+      }
+    }
+    const bodyData = JSON.stringify(request)
+  
+    const response = await fetch(url, {
+      method: 'post',
+      headers: await getJiraRequestHeader(),
+      body: bodyData,
+    })
+
+    .then(restErrorResponseHandler)
+  
+    .then( async response => {
+      return {requestOk: response.ok, statusText: response.statusText}
+
+    });
```

Comment:
doesn't this endpoint return a json body? if so, it's more meaningful to return `response.json()`, because `response.ok` and `response.statusText` are pretty much just http implementation details. The graphql type can just expose a subset of the data returned if it's a lot of uninteresting data, but something meaningful is still better :)

Reply:
I tried that initially and noticed that for failed attempts ( i.e. unauthenticated) the returned json body would be considered invalid, and there would be a parsing error thrown

Reply:
this is the authentication of our app against JIRA right? If so this error represents a bug in the system rather than a meaningful error that user should be expected to handle.

The best thing to do there would be to test the credential at the app's initialization time and fail the app if the credential isn't correct. In the elasticsearch client case, we do this by pinging elasticsearch using the instantiated client, if the ping fails then we know there's a bug in the configuration and the app should not be able to start up.